students/shirin_ak/Session04/mailroom.py

```diff
@@ -0,0 +1,225 @@
+        to the donation history """
+
+    while True: 
+
+        print("\nEnter 'list' to see list of names.") 
+
+        print("Enter 'quit' to go back.")
+        print("Enter 'name' to add new donor.")
+        
+        name = input("Please enter a name: ")
+        
+        if (name == 'list'):
+ 
+            Show_list()
+            
+        elif (name == 'quit'): 
+            break
+        
+        else: 
+
+            donation_amount = get_donation() 
+
+            if (donation_amount == 'quit'):
+ 
+                break
+            
+            if (name not in Donors):
+                
+                Donors[name] = [1, donation_amount]
+                
+            else:
+                
+                Donors[name][0] = Donors[name][0] + 1 
+ 
+                Donors[name][1] = Donors[name][1] + donation_amount
+ 
+            Send_Thank_you_letter(name, donation_amount)
+ 
+            break
+        
+        
+def Show_list():
+    
+    print("\nDonor Names:") 
+    print("-" *15 ) 
+    for name in Donors:
+ 
+        print(name)       
+
+
+def get_donation():
+    """ Take donation from the donor"""
+
+    while True: 
+        print("\nPlease enter donation amount. " 
+
+             " Or Enter 'quit' to return to main menu.")
+ 
+        donation = input("$")
+        
+        if (donation == 'quit'): 
+            return donation
+        
+        else:
+            donation = float(donation)
+ 
+            return donation 
+
+            
+              
+
+def Send_Thank_you_letter(name, donation_amount): 
+    message = ("\nDear {},\n\nThank you for your most recent donation of ${}. " 
+
+               " You have donated\na total of ${} to the " 
+               "BEYOND VISION MUSIC FOUNDATION.  Our company greatly appreciates the continued " 
+
+               "generosity from individiuals like you.")
+               
+
+    message = (message.format(name, donation_amount, Donors[name][1])) 
+#    print(message) 
+    return message
+
+
+
+def Save_letter(letter, name, amount): 
+ 
+    ''' 
+    Save a copy of a Thank You letter to the local disk when the letter is sent 
+    ''' 
+    file_name = '{}_{}'.format(name.replace(' ', '_'), amount.replace('.', '_')) 
+    f = open(file_name, 'w') 
+    f.write(letter) 
+    f.close() 
+
+
+def Send_letter_for_all(): 
+
+    for name in list(Donors.keys()): 
+
+        amount = Donors[name][0] * Donors[name][1] 
+        message = Send_Thank_you_letter(name, amount) 
+        Save_letter(message, name, str(amount))
+      
+
+
+def exit_program():
+    
+    print("Bye!")
+    sys.exit(0)
+              
+
+
+def menu_option(): 
+    """ 
+    Print out the main application menu and then read the user input. 
+    """ 
+    action = input(('''Please choose a menu option below:
+
+      1 - Send a Thank You 
+      2 - Create a Report 
+      3 - Send letters to everyone 
+      4 - Quit 
+      >>''')) 
+
+    return action.strip() 
```

Comment:
why the `.strip()` here?  To remove newline?  You could just cast to `int`
```
action = int(input(('''Please choose a menu option below:
      1 - Send a Thank You 
      2 - Create a Report 
      3 - Send letters to everyone 
      4 - Quit 
      >>''')))
```
And then have your keys in your main be integers instead of strings.